### ISSUE TITLE
'SIGALRM' undeclared when installing on Windows

### ISSUE THREAD

**Issue by RaisssHab** [micans/mcl#9]
Hello. I'm using cygwin on Windows to install the MCL. 

The first problem I encountered was 'checking build system type... autofoo/config.guess: unable to guess system type'. I solved it by replacing 'autofoo/config.guess' by a new one from https://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD.

After configuring I tried to run 'make install' command and got the following:

> proc.c: In function 'mclSigCatch':
> proc.c:65:18: error: 'SIGALRM' undeclared (first use in this function)
>     {  if (sig == SIGALRM)
>                   ^
> proc.c:65:18: note: each undeclared identifier is reported only once for each function
>  it appears in
> make[2]: *** [Makefile:239: proc.o] Error 1
> make[2]: Leaving directory '/cygdrive/c/Users/raish/Downloads/22.08.2022 Работа/mcl-14-137/src/mcl'
> make[1]: *** [Makefile:233: install-recursive] Error 1
> make[1]: Leaving directory '/cygdrive/c/Users/raish/Downloads/22.08.2022 Работа/mcl-14-137/src'
> make: *** [Makefile:279: install-recursive] Error 1

How can I solve this problem?

**Comment by micans** [comment]
Hi, thanks for the bug report. This is due to SIGALRM not existing in cygwin. As a short term fix, you can change the code around line 65 in the file `src/mcl.c` to this: (so substitute SIGQUIT for SIGALRM).

```
void mclSigCatch
(  int sig
)
   {  if (sig == SIGQUIT)
      abort_loop = 1
;  }
```

I haven't tested this (no access to cygwin currently), but think this should work.

The purpose of this bit of this bit of code is to enable users to signal mcl if it encounters a rare flip flop state where expansion and inflation cancel each other out - with large networks mcl might keep running for a long time. In the upcoming release this provision will no longer be needed, as mcl will enforce convergence after a certain number of iterations. Nevertheless, I will probably seek to just add some cygwin-specific test to fix this.

Thanks as well for flagging `config.guess`.

**Comment by RaisssHab** [comment]
Hello. Unfortunately, replacing SIGALRM with SIGQUIT doesn't work as well. The error is the same as for SIGALRM.

**Comment by RaisssHab** [comment]
Okay, the problem was with that I didn't install gcc for cygwin. The problems stated here do not appear anymore (including SIGALRM). However, there's another problem, which I'm going to describe in another issue.
